### ISSUE TITLE
General performance issues - Kiwi is using approx 100% CPU in browser task manager.

### ISSUE THREAD

**Issue by AltitudeApps** [kiwiirc/kiwiirc#1665]
I hesitate to post this, because I can't really identify an actual bug.

But over the last few weeks I have noticed that Kiwi has become sluggish, so much so that clicking on the message compose area won't display the reticle or accept text input for ten seconds.  I have also experienced many occurrences of Kiwi simply losing its connection.  Yup, it just happened again as I glance at that screen...

I have been using Kiwi happily for years without any trouble.

Is there something new in Vue that might be causing massively increased CPU usage?

Maybe it is one of the new security features being rolled out in MS Edge which is going bananas?

I am at my wit's end trying to diagnose this.  Is anyone else reporting similar troubles?

@prawnsalad 

P.S.  I'm also seeing this happen on Safari on iPad.  I leave my iPad open with Kiwi in the foreground, just to keep an eye on activity as I walk around the house.  I've been doing this for years as well.  But recently it just loses its connection all the time.

**Comment by ItsOnlyBinary** [comment]
the disconnecting, is likely the mobile browser pausing all javascript (this will include tablets) to save power.

as for 100% cpu usage, its a little more tricky...
are you using your own kiwiirc instance or kiwiirc.com?
are there any plugins loaded?
have you increased messages in scrollback higher than 250?
what cpu?

i've not had any recent reports of cpu issues

**Comment by AltitudeApps** [comment]
TL;DR:  You are right, my scrollback was set to 2250 instead of 250.
----

I am using kiwiirc.com   Connecting to irc.zeronode.net 

no plugins

I will check my scrollback settings.  I recently increased the scrollback, but my troubles were happening before that.  (But they have gotten worse lately, I'll try reducing the scrollback).

Incidentally, the cpu usage on the cpu itself was greater than 50% for the thread where the Kiwi tab lived.

As far as the mobile browser disconnecting, I am aware of that phenomenon, but in this case it doesn't apply.  As long as the Kiwi tab is in the foreground and the iPad doesn't turn the screen off, Kiwi happily sits there updating the room activity to the screen.  (I will check to see if I raised the scrollback about 250 on the iPad, too; I might have got fed up with missing messages, and then configured the scrollback on all my endpoints.)

So, I will check my scrollback on the iPad...    Hmm, OK, it is set at 250.   But, I think that my troubles on the laptop might have clobbered by WiFi connection, which caused the iPad to lose its WiFi connection.  So the problem on my iPad might be a secondary consequence of what is happening on the laptop.

Whoops, scrollback on my laptop is set to 2250.  No wonder.

However, I have used a scrollback larger than 250 before, and the performance was tolerable.  But, it might have been about 1000, not 2250.
